### DISCUSSION TITLE
本科毕业论文参考文献一直报错

### DISCUSSION THREAD

**Discussion by Yu-steve** [tuna/thuthesis#873] [Q&A] [answered]
我使用了Texstudio进行编译，在引用参考文献的时候一直报错，请问该怎么解决啊

我在main.tex里面设置了
\bibliography{ref/refs}  % 参考文献使用 BibTeX 编译

在thusetup里面设置了
% 本科生参考文献的著录格式
 \usepackage[sort]{natbib}
 \bibliographystyle{thuthesis-bachelor}
% \usepackage[backref]{hyperref}
 \bibliography{{ref/refs}}

结果报错
Missing \begin{document}. \begin{thebibliography}{8}
Undefined control sequence. \begin{thebibliography}{8}

百度上说可能少了宏包，但也找不到该加什么东西。

![1](https://user-images.githubusercontent.com/82199782/236677509-dc51931b-baf3-42f9-8b37-74f8263e916b.png)
![3](https://user-images.githubusercontent.com/82199782/236677513-56bcf987-a17e-494e-868a-cb63034be1
![2](https://user-images.githubusercontent.com/82199782/236677516-70e7cf56-7cd4-4b00-80a2-266d1df411d2.png)
27.png)

**Comment by Yu-steve** [comment]
![3](https://user-images.githubusercontent.com/82199782/236677610-8df09403-b985-4663-be3b-f3caed61141b.png)
还有这张图，thusetup里面

**Comment by note286** [comment]
请给出能够复现问题的代码，从`\documentclass`到`\end{document}`都要。移除所有的不必要的宏包，尽可能减小代码量，并适当地缩进和换行。 此外，要说清楚编译命令，编译平台和发行版版本等信息，没人关心你用了什么编辑器，说清楚编译命令即可。

代码不要截图，请使用markdown语法给出。

**Reply by Yu-steve** [reply, marked as answer]
谢谢学长，刚才我尝试复现的时候，发现是ref.bib坏了，我把它替换之后就不会报错了，参考文献能够正常读取并显示

**Comment by zepinglee** [comment]
`\bibliography{}` 应该在输出参考文献表的位置，而不是导言区。

**Reply by Yu-steve** [reply]
感谢，后面好像没问题了，可能在thuthesis.cls文件里自带这行代码吧

**Comment by zepinglee** [comment]
示例文档的 `\bibliography{}` 在这里

https://github.com/tuna/thuthesis/blob/e26380bf6979924acfc74819a2faf5b3ab721d48/thuthesis-example.tex#L62-L63

不在这里

https://github.com/tuna/thuthesis/blob/e26380bf6979924acfc74819a2faf5b3ab721d48/thusetup.tex#L157-L160
